django/db/migrations/recorder.py

```diff
@@ -47,16 +47,21 @@ def __str__(self):
 
     def __init__(self, connection):
         self.connection = connection
+        self._has_table = False
 
     @property
     def migration_qs(self):
         return self.Migration.objects.using(self.connection.alias)
 
     def has_table(self):
         """Return True if the django_migrations table exists."""
-        with self.connection.cursor() as cursor:
-            tables = self.connection.introspection.table_names(cursor)
-        return self.Migration._meta.db_table in tables
+        if not self._has_table:
+            # False, re-check
+            with self.connection.cursor() as cursor:
+                tables = self.connection.introspection.table_names(cursor)
+
+            self._has_table = self.Migration._meta.db_table in tables
+        return self._has_table
```

Comment:
Not quite, because we only want to cache a positive result rather than a negative. If the table doesn't exist then we want to keep checking, but if it _does_ exist then we can just keep returning `True`